### DISCUSSION TITLE
Officially allow projects with copyleft licenses to join the dotnet foundation

### DISCUSSION THREAD

**Discussion by wastaz** [dotnet-foundation/Home#78] [Board of Directors]
I noticed when perusing some issues in the dotnet foundation trackers that apparently projects using copyleft licenses such as GPL are not allowed to join the dotnet foundation since they are not licensed under a "permissive" open source license. And honestly, this leaves quite a bad taste in my mouth.

Given the discussions over the last couple of years of open source development, who pays for it and even maintainer burnout I have personally come to the conclusion that the most "ethical" commonly used open source licenses today are in fact the copyleft ones (with or without dual licensing to allow for closed source commercial use). 

There are no shortage of examples of everything from small companies to multinational giants basing huge parts of their products and their work on open source projects. In the best case actually following the licenses and providing a link back to the projects in their readmes or documentation, but extremely rarely actually supporting the projects in question with resources or even code. Open source maintainers, often using their own spare time to run their projects, may occasionally get some time in the spotlight if their project grows big. But they also tend to get a lot more unpaid work and maintenance, some in the not so very fun form of angry issues about bugs that they "absolutely must fix immediately" because some company who has never even contributed to the project depends on it.

Copyleft licenses at least "force" some sort of payment for services rendered. The payment may be in form of other code that also becomes open sourced, or it may come in form of income from dual licensing. **While it doesn't completely solve the issue**, it at least equalizes the playing field a little bit.

Copyleft licenses also **has the potential** to actually create healthier open source projects than projects with "permissive" licenses. Since it forces sharing of code it at least in theory should force more cooperation and sharing towards the project, improving the quality of code for everyone. 

In either way, even though copyleft licenses are not the perfect solution to all problems and even though they are not an especially popular form of licensing in the dotnet community at large. It makes zero sense to exclude copyleft licenses from membership in the dotnet foundation.

I urge the board of directors to reconsider the exclusion of copyleft licenses from your project charter. It honestly feels mostly like an antique remnant from the old days when Microsoft fought against open source instead of embracing it.

**Comment by robmen** [comment]
This sounds very much like what I wrote a couple of years ago: [The decline of reciprocal licenses through corporate leadership](https://robmensching.com/blog/posts/2020/7/16/the-decline-of-reciprocal-licenses-through-corporate-leadership/).

FWIW, the [WiX Toolset](https://wixtoolset.org/) (the project I lead) uses a reciprocal license for most of the reasons above (but not dual-licensed).
